### PR TITLE
fix(wallet): Sort Portfolio Groups by Fiat Value

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Will now sort Portfolio groups `Accounts` or `Networks` by `Fiat Value`

Resolves <https://github.com/brave/brave-browser/issues/31024>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Go to the `Portfolio` page and group your portfolio by `Accounts`
2. Account groups should be sorted by `Fiat Value` high to low
3. Group your portfolio by `Networks`
4. Network groups should be sorted by `Fiat Value` high to low

Before:

https://github.com/brave/brave-core/assets/40611140/d58ff0c1-8112-4d3a-a45f-805ffef29ef4

After:

https://github.com/brave/brave-core/assets/40611140/03fb1c83-0a54-460f-ade3-ce351b4a4000

### CODE REVIEW

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx [hunk 1]
```diff
@@ -231,13 +231,12 @@ export const TokenLists = ({
       // Return an empty string to display a loading
       // skeleton while assets are populated.
       if (sortedFungibleTokensList.length === 0) {
-        return ''
+        return new Amount('')
```
Comment (onyb) [inline components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx:234]:
```suggestion
        return Amount.empty()
```

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx [hunk 2]
```diff
@@ -256,13 +255,11 @@ export const TokenLists = ({
 
       return !reducedAmounts.isUndefined()
         ? reducedAmounts
-          .formatAsFiat(defaultCurrencies.fiat)
-        : ''
+        : new Amount('')
```
Comment (onyb) [inline components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx:258]:
```suggestion
        : Amount.empty()
```

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx [hunk 3]
```diff
@@ -339,21 +335,27 @@ export const TokenLists = ({
...
-    defaultCurrencies.fiat,
     sortedFungibleTokensList
   ])
 
   const listUiByNetworks = React.useMemo(() => {
-    return networks?.map((network) =>
+    return networks?.sort((a, b) => {
+      const aBalance = getNetworkFiatValue(a)
+      const bBalance = getNetworkFiatValue(b)
+      return bBalance.toNumber() - aBalance.toNumber()
```
Comment (onyb) [inline components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx:348]:
```suggestion
      return bBalance.minus(aBalance).toNumber()
```

**Review by onyb — APPROVED** [review]
Left some minor comments. Thanks for doing this! 🙏

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx (hunk 1 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx]: Updated!

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx (hunk 2 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx]: Updated!

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx (hunk 3 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx]: Updated!

**Pull request merged**